### DISCUSSION TITLE
Possible to add SortDirection as an optional parameter to SortingFn?

### DISCUSSION THREAD

**Discussion by rupayan-ninety-eight** [TanStack/table#5453] [Ideas]
Something like this: 

`export type SortingFn<TData extends AnyData> = {
  (rowA: Row<TData>, rowB: Row<TData>, columnId: string, sortDirection: SortDirection): number
}`

A table component I'm working on needed `null` values to be placed last in the list regardless of sorting order. I tried using `sortUndefined` but it doesn't seem to work with custom sorting functions.

I had to then try a workaround, where I used the sorting state from `const [sorting, setSorting] = useState([])` to figure out the sort order within my custom sorting function.

If I'm missing an easier way to do this, please let me know.

**Comment by magnimarels** [comment]
How I approach this problem in my project is by setting a SortingState

`const [sorting, setSorting] = useState<SortingState>([
        {
            desc: isDesc ? true : false
        }
    ])`
    
and then passing it into my tables initial state
    
`const table = useReactTable({ initialState: { sorting: sorting } })`

along with the other options
